### ISSUE TITLE
Issue with trajectory prediction

### ISSUE THREAD

**Issue by sharmasushil** [alfredgu001324/MapUncertaintyPrediction#29]
Hi,

Thank you very much for your help. 

I've encountered an issue: when I run the visualization, I can only see GT, with no changes in the trajectory prediction. 

I can share some results using HiVT and DenseTNT if that would be helpful.
Below is HiVT
<img width="810" alt="Screenshot 2024-11-04 at 12 18 54" src="https://github.com/user-attachments/assets/42558e84-4bab-41fc-be86-e19802d9306a">
The below one is DenseTNT:
<img width="358" alt="Screenshot 2024-11-04 at 12 22 24" src="https://github.com/user-attachments/assets/3c7fbe94-621a-4a6f-817a-701f6ef3f60a">

Please let me know where I am doing mistake 

Thanks a million 

Your were very helpful so far

**Comment by alfredgu001324** [comment]
May I know how you run the visualization? You need to save the trajectory prediction results, and then load them in the visualization step. You can take a look at the visualization part: https://github.com/alfredgu001324/MapUncertaintyPrediction/blob/main/docs/trj.md

**Comment by sharmasushil** [comment]
Hi, 
Thank you for your reply.

I am using to save pickle file for trajectory (base and unc) using below command:
python eval.py --root ../trj_data/maptrv2 --split mini_val --method base  --batch_size 32 --ckpt_path /media/xxx/Storage/MapUncertaintyPrediction/HiVT_modified/epoch=63-step=411903.ckpt

then I do visualization:

 python vis_std.py --version mini --dataroot ../nuscenes --split mini_val --trj_pred HiVT --map MapTR --trj_data ../trj_data/maptrv2/mini_val/data --base_results /media/xxx/Storage/MapUncertaintyPrediction/HiVT_modified/result_base.pkl --unc_results /media/xxx/Storage/MapUncertaintyPrediction/HiVT_modified/result_unc.pkl --boxes /media/xxx/Storage/MapUncertaintyPrediction/bbox.pkl --save_path /media/xxx/Storage/MapUncertaintyPrediction/Results


this is the step, I am following 


Thanks

**Comment by sharmasushil** [comment]
Just wondering if this is how you save results.pickle for unc and base:

https://github.com/alfredgu001324/MapUncertaintyPrediction/blob/main/HiVT_modified/models/hivt.py#L202

**Comment by alfredgu001324** [comment]
Yes correct, you need to uncomment this line: https://github.com/alfredgu001324/MapUncertaintyPrediction/blob/08baf647b8fb7d20cb2840ed8996dd788cd38f3f/HiVT_modified/models/hivt.py#L138 to save the trajectories.

Sorry it is indeed an ugly solution....

**Comment by sharmasushil** [comment]
I have uncommented this line, but I still have the same issue with visualization as above I mentioned.
I don't know where I am making a mistake even following everything step by step.

I believe the model is not predicting the trajectory of HiVT, even DenseTNT has some predictions. 

Please advise on this

**Comment by sharmasushil** [comment]
Hi, 

sample trajectory which you shared doesn't work. 
and give me error: 

  File "vis_std.py", line 291, in main
    x, y = data[int(key)]['ego_pos'][0, 0].item(), data[int(key)]['ego_pos'][0, 1].item()
KeyError: 614

for visualization:
--base_results ../PATH_to_baseline_prediction_results \
  --unc_results ../PATH_to_unc_prediction_results \


Please advice me on this. 

Thanks a million

**Comment by alfredgu001324** [comment]
Are you loading the current splits? The sample trajectories I shared is for the entire val split, which contains 4519 scenarios. If you see a key error then maybe check L249?

**Comment by sharmasushil** [comment]
Hi, 

Thank you so much for your help. 
I can successfully implement the DenseTNT model for trajectory prediction, but I'm having an issue as I show your output. The prediction makes sense but my case trajectory shows:
![MicrosoftTeams-image (160)](https://github.com/user-attachments/assets/8b795d12-758f-44b9-9016-943091ec2081)


Please let me know if it looks normal (Pink dotted in prediction and red is GT)

Thanks !

**Comment by alfredgu001324** [comment]
Hi, sorry I did not quite get your question. Can you explain again? Thank you!

**Comment by flotaas** [comment]
Hi,
I've encountered a similar issue when running the visualization.  The pink predicted trajectory completely overlaps with the red GT. Below is the command I ran “python vis_std.py 
--version trainval 
--dataroot ../Nuscenes/ 
--split val 
--trj_pred HiVT 
--map MapTR 
--trj_data ../trj_data/maptrv2/val/data 
--base_results /MapUncertaintyPrediction/HiVT_modified/val_base_result.pkl 
--unc_results /MapUncertaintyPrediction/HiVT_modified/val_unc_result.pkl 
--boxes /MapUncertaintyPrediction/MapTRv2_modified/bbox.pkl 
--save_path /MapUncertaintyPrediction/MapTRv2_modified/Results”, where val_base_result.pkl and val_unc_result.pkl were obtained using python eval.py. 



However, when I run the command using sample trajectories files (https://drive.google.com/drive/folders/161c5PH_ccZYi2wj0MnLU2KJCDzwAyCSt?usp=sharing) shared by the author, it works normally. 



It seems that there is an issue with the obtained trajectory .pkl file, but I generated it by running the provided script https://github.com/alfredgu001324/MapUncertaintyPrediction/blob/main/docs/trj.md. Why is this happening?

**Comment by alfredgu001324** [comment]
@flotaas Thanks for reaching out. Unfortunately I am not too sure either, maybe can you check whether the metrics is actually being calculated first? At least just to make sure the trajectories that are being saved are the actual predicted trajectories. There should be 6 trajectories being saved.

**Comment by alfredgu001324** [comment]
@flotaas Hi, can you try the new vis function I just updated? It now saves all the vehicles trajectories for each scene and group them accordingly. It should give you the correct pkl format, and the vis script should work.

**Comment by flotaas** [comment]
> [@flotaas](https://github.com/flotaas) [@sharmasushil](https://github.com/sharmasushil) Hi, can you try the new vis function I just updated? It now saves all the vehicles trajectories for each scene and group them accordingly. It should give you the correct pkl format, and the vis script should work.

Thank you !

**Comment by Amireux52** [comment]
> Hi,
> 
> sample trajectory which you shared doesn't work. and give me error:
> 
> File "vis_std.py", line 291, in main x, y = data[int(key)]['ego_pos'][0, 0].item(), data[int(key)]['ego_pos'][0, 1].item() KeyError: 614
> 
> for visualization: --base_results ../PATH_to_baseline_prediction_results --unc_results ../PATH_to_unc_prediction_results \
> 
> Please advice me on this.
> 
> Thanks a million
您好，我遇到同样的问题，请问应该怎么解决哇，盼复，感谢您
